### ISSUE TITLE
Failure to resolve to long filename in dependency walker output

### ISSUE THREAD

**Issue by 123GitHub0123** [Nuitka/Nuitka#2514]
Before submitting an Issue, please review the
[Issue Guidelines](https://github.com/Nuitka/Nuitka/blob/develop/CONTRIBUTING.md#submitting-an-issue).

- Please check whether the bug was already reported or fixed.

- Please check out if the develop version of Nuitka works better for you.

  Download source, packages [from here](https://nuitka.net/doc/download.html) where you will also
  find instructions how to do it via pip or git.

If you want to post a problem/bug, to help us understand and resolve your issue please check that
you have provided at least the information below, and discard up to here:

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version

  If you are not providing the full output. The issue cannot be solved and will be considered
  `invalid`. The command outputs more than you think, and we are adding more all the time. We do not
  want to left guessing or trying things out to reproduce your issue.

- How did you install Nuitka and Python

  Did you use pip, anaconda, deb, rpm, from source, git clone, then install into a virtualenv or
  not, this is very important usually and one thing, the above command does not tell us (yet).

- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip freeze

- Many times when you get an error from Nuitka, your setup may be special

  Then even a `print("hello world")` program will not work, please try that and report that error
  instead. We do not need a report that says "Tensorflow does not work" when nothing actually works
  for you. You ought to also read the User Manual and check that your setup is actually supported
  already.

- Also supply a Short, Self Contained, Correct, Example

  That demonstrates the issue i.e a small piece of code which reproduces the issue and can be run
  with out any other (or as few as possible) external dependencies. Issues without this may will
  likely be rejected without much consideration. Often this can be as simple as importing a package,
  if this is a packaging issue, try that first.

  Pointers to repositories with usage of `pipenv` are very welcome, archives with examples are too,
  e.g. if a certain package structure is needed. This must be only source code, binaries are not
  used ever by us. But beware, that e.g. we cannot click around and stuff. Still do an effort to
  make the usage obvious. Having a compile script in the repo is perfect.

  But this cannot be much code for non-commercial users, since that causes too much effort. We
  cannot just compile your code, run it and have it download a control and command center.

- Provide in your issue the Nuitka options used

  Ideally use the `# nuitka-project:` options feature in the code, so options and example code go
  along. Alternatively state the command line.

  [Nuitka Options in the code](https://nuitka.net/doc/user-manual.html#nuitka-options-in-the-code)

- Avoid unnecessary options

  Do not use `--onefile` if the issue also happens with `--standalone`. Minimize the use of options
  as far as you can, please. Do not disable outputs with `--quiet` and do not disable warnings.

- Note if this is a regression

  If it used to work in an earlier version of Nuitka, please note what you know about that. Since
  git bisect is a thing for which we do a lot of error to make it usable, this will help
  dramatically to isolate the issue.

- Consider getting commercial support

  [Nuitka commercial](https://nuitka.net/doc/commercial.html) offers subscriptions and priority
  support. This will accelerate your problem solution and helps to sustain Nuitka development. Also
  you then have a chance to provide source code of your project to us, which might simplify things,
  or pay for time to solve your issues in your environment.

Some things are not welcome, please consider it.

- Do *not* post screenshots

  These are not welcome unless absolutely necessary, e.g. because of Qt display problem, instead
  capture the output of programs, so things are searchable and copy&paste will work. I just plainly
  don't want to manually copy strings and hope they match.

- Do *not* close the issue yourself, we will close things on stable releases

  We close issues only when they are released as a stable version, e.g. in a hotfix or a new
  release, before that it will be "Done" in planning and go through `factory` and `develop` tags to
  indicate they are solved there.

  Of course, if you find out your issue is invalid, please do close it, and we then attach the
  `invalid` tag.

- Do *not* report against factory version

  Unless you were asked to test it there, it is frequently very broken, and there is only noise to
  be had. Telling me about it on Discord would be a better idea.

- Do *not* let this template remain part of the issue, it's noise.

**Comment by 123GitHub0123** [comment]
<nuitka-compilation-report nuitka_version="1.8.4" nuitka_commercial_version="not installed" completion="exception">
  <exception exception_type="OSError" exception_value="[Errno 123] &#25991;&#20214;&#21517;&#12289;&#30446;&#24405;&#21517;&#25110;&#21367;&#26631;&#35821;&#27861;&#19981;&#27491;&#30830;&#12290;">

**Comment by 123GitHub0123** [comment]
python -m nuitka --standalone   --mingw64 -o main --output-dir=dist --show-progress  --windows-icon-from-ico=out3.ico main.py

**Comment by 123GitHub0123** [comment]
<nuitka-compilation-report nuitka_version="1.8.4" nuitka_commercial_version="not installed" completion="exception">
  <exception exception_type="OSError" exception_value="[Errno 123] &#25991;&#20214;&#21517;&#12289;&#30446;&#24405;&#21517;&#25110;&#21367;&#26631;&#35821;&#27861;&#19981;&#27491;&#30830;&#12290;"

**Comment by 123GitHub0123** [comment]
nuitka-compilation-report nuitka_version="1.8.4" nuitka_commercial_version="not installed" completion="exception">
  <exception exception_type="OSError" exception_value="[Errno 123] &#25991;&#20214;&#21517;&#12289;&#30446;&#24405;&#21517;&#25110;&#21367;&#26631;&#35821;&#27861;&#19981;&#27491;&#30830;&#12290;"

**Comment by 123GitHub0123** [comment]
*console_path* D:\bin>

**Comment by 123GitHub0123** [comment]
It's strange that when I try a few more times, I can succeed once.

**Comment by kayhayen** [comment]
Please next time try to fill out the issue template properly, you are expected to submit it when complete. 

I did two things, first I am adding the filename to the output of the error message, so it says which filename gives it a hard time. Second I think I found a likely cause, namely that it's try to do it for the main program filename that it otherwise ignores in dependency walker output, and that's likely the trouble maker, I am sure it might output strange things for the unicode input path there.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback if this is working is very welcome, just please do not share plans of doing it, but rather confirmations or denials of it working.

**Comment by kayhayen** [comment]
This is part of the hotfix release 1.8.5 that I just made.

**Comment by ShannonZ** [comment]
`Nuitka:INFO: Generating source code for C backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.                    
Nuitka:INFO: Running C compilation via Scons.
Nuitka-Scons:INFO: Backend C compiler: cl (cl 14.3).
Nuitka-Scons:INFO: Backend linking program with 549 files (no progress information available for this stage).
Nuitka-Scons:INFO: Compiled 549 C files using clcache with 474 cache hits and 75 cache misses.
Nuitka-Postprocessing:INFO: Adding 1 icon(s) from icon file '.\icon.ico'.
Nuitka-Plugins:INFO: data-files: Included data file 'grpc\_cython\_credentials\roots.pem' due to package data for 'grpc'.
Nuitka-Reports:INFO: Compilation report written to file 'nuitka-crash-report.xml'.
Detecting used DLLs: 0.8%|▏                        | 1/126, main.exeTraceback (most recent call last):
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\__main__.py", line 170, in <module>
    main()
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\__main__.py", line 156, in main
    MainControl.main()
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\MainControl.py", line 1116, in main
    _main()
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\MainControl.py", line 1008, in _main
    detectUsedDLLs(
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\freezer\Standalone.py", line 335, in detectUsedDLLs
    _detectUsedDLLs(
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\freezer\Standalone.py", line 253, in _detectUsedDLLs
    used_dlls = _detectBinaryDLLs(
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\freezer\Standalone.py", line 142, in _detectBinaryDLLs
    return detectBinaryPathDLLsWin32(
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\freezer\DllDependenciesWin32.py", line 97, in detectBinaryPathDLLsWin32
    result = detectDLLsWithDependencyWalker(
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\freezer\DependsExe.py", line 234, in detectDLLsWithDependencyWalker
    result = parseDependsExeOutput(output_filename)
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\freezer\DependsExe.py", line 175, in parseDependsExeOutput
    return _parseDependsExeOutput2(getFileContentByLine(filename, encoding="latin1"))
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\freezer\DependsExe.py", line 149, in _parseDependsExeOutput2
    dll_filename = getWindowsLongPathName(dll_filename)
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\utils\FileOperations.py", line 1150, in getWindowsLongPathName
    raiseWindowsError("getWindowsLongPathName for %s" % filename)
  File "D:\Projects\Python\gRPCServer\VirtualEnv\venv\lib\site-packages\nuitka\utils\Utils.py", line 406, in raiseWindowsError
    raise WindowsError(
OSError: [Errno 123] getWindowsLongPathName for d:\programs\web\dll\ws2_32.dll 
Nuitka-Reports:INFO: Compilation crash report written to file 'nuitka-crash-report.xml'. Please include it in your bug report.`

**Comment by ShannonZ** [comment]
Encountered the same problem with version 1.9.7

**Comment by ShannonZ** [comment]
[nuitka-crash-report.xml.txt](https://github.com/Nuitka/Nuitka/files/14035013/nuitka-crash-report.xml.txt)

**Comment by kayhayen** [comment]
Posting below closed issues is a surefire way to get ignored. I am not going to guess without a fresh issue template, what kinds of mistakes and setup you have made, and I am not going to trace, or remember to check later, or searching for where something was said. Please open new issues for new issues. If the issue is closed, it's not the same even if the symptom looks similar to you. @ShannonZ
